sys/tools/syscalls/freebsd-syscall.lua

```diff
@@ -0,0 +1,134 @@
+--
+-- SPDX-License-Identifier: BSD-2-Clause
+--
+-- Copyright (c) 2023 Warner Losh <imp@bsdimp.com>
+-- Copyright (c) 2024 Tyler Baxter <agge@FreeBSD.org>
+--
+
+-- Derived in large part from makesyscalls.lua:
+--
+-- SPDX-License-Identifier: BSD-2-Clause-FreeBSD
+--
+-- Copyright (c) 2019 Kyle Evans <kevans@FreeBSD.org>
+
+local syscall = require("syscall")
+
+local FreeBSDSyscall = {}
+
+FreeBSDSyscall.__index = FreeBSDSyscall
+
+-- Processes compatability options in the global config and inserts them into
+-- known_flags to reference.
+function FreeBSDSyscall:processCompat()
+	for _, v in pairs(config.compat_options) do
+		if v.stdcompat ~= nil then
+			local stdcompat = v.stdcompat
+			v.definition = "COMPAT_" .. stdcompat:upper()
+			v.compatlevel = tonumber(stdcompat:match("([0-9]+)$"))
+			v.flag = stdcompat:gsub("FREEBSD", "COMPAT")
+			v.prefix = stdcompat:lower() .. "_"
+			v.descr = stdcompat:lower()
+		end
+
+		-- Add compat option to syscall.known_flags
+	    table.insert(syscall.known_flags, v.flag)
+	end
+end
+
+function FreeBSDSyscall:parseSysfile()
+	local file = self.sysfile
+	local config = self.config
+	local commentExpr = "^%s*;.*"
+
+    -- Keep track of the system call numbers and make sure there's no skipped 
+    -- system calls.
+    local num = 0
+
+	if file == nil then
+		print "No file"
+		return
+	end
+
+	self.syscalls = { }
+
+	local fh = io.open(file)
```

Comment:
Ditto note about preserving the failure msg for the failure note below.  Even better here might be converting all of these print + return {} combo to return `nil, <msg>` on failure to avoid the library printing things that the frontend doesn't want and give the consumer control + ability to use the standard assert() pattern.

Reply:
Yes! These have been changed. The library should not be printing anything. The other suggestions in this thread have also been incorporated. Thank you Kyle!